src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.io.*;
+
+import duke.task.*;
+import duke.command.TaskParser;
+
+public class Duke {
+
+    static ArrayList<Task> taskList = new ArrayList<>();
+    static final String FILENAME = "tasklist";
+
+    private static void saveTaskListToFile() {
+	try {
+	    FileOutputStream fileos = new FileOutputStream(FILENAME);
+	    ObjectOutputStream objos = new ObjectOutputStream(fileos);
+	    objos.writeObject(taskList);
+	} catch (IOException e) {
+	    System.out.println("Error saving to file.");
```

Comment:
Error strings are also considered as constants and should be added as named constants